proto/streamlit/proto/BackMsg.proto

```diff
@@ -73,6 +73,9 @@ message BackMsg {
 
     // Sends an app heartbeat message through the websocket
     bool app_heartbeat = 17;
+
+    // Sends the st.context required info to the server for the current session
+    ContextClientInfo context_client_info = 18;
```

Comment:
Have you considered directly adding it to `ClientState` and handling it exactly the same as query parameters? I think this information should be up-to-date and available at the beginning of every rerun and sending it in ClientState with every rerun seems reasonable